router/vendor.go

```diff
@@ -66,7 +66,7 @@ func NewVendorEngine(
 	})
 
 	r.GET(cfg.GetById, func(ctx *gin.Context) {
-		id := ctx.Request.PathValue("id")
+		id := ctx.Param("id")
```

Comment:
Could you please fix issues like this in `./router/product.go` for `UpdateProduct` and `UpdatePrice`?